fairscale/utils/reduce_scatter_bucketer.py

```diff
@@ -122,9 +128,16 @@ def reduce_scatter_async(
 
         bucket_shard_size = self._get_shard_size(first_input.element_size(), world_size)
         if first_input_size > bucket_shard_size:
+            # TODO: investigate how to avoid using torch.cat
```

Comment:
```suggestion
            # TODO: investigate how to avoid using torch.cat (because it seems to be slow for CPU tensors)
```

Reply:
I don't think this was about CPU tensors. torch.cat results in 1 tensor copy which we can avoid if we manipulate the tensors in place.